iphone/Classes/TiUITableViewProxy.m

```diff
@@ -882,9 +882,9 @@ - (void)willShow
   [(TiUITableView *)[self view] refreshSearchControllerUsingReload:YES];
 }
 
-- (NSUInteger)sectionCount
+- (NSNumber *)sectionCount
 { //TODO: Shouldn't this be in the main thread, too?
-  return [sections count];
+  return NUMUINTEGER((sections != nil) ? sections.count : 0);
```

Comment:
I think this is the only required change, and the rest of the lines/changes are not necessary. Why not just leave this method as returning `NSUInteger` as before and avoid the need to explicitly call `[NSNumber unsignedIntegerValue]` to get one in the above changes?

Reply:
It wasn't crashing because `sections` was `nil`.
_(That's just a safety mechanism I added just in case.)_

It was crashing because the calling code [here](https://github.com/appcelerator/titanium_mobile/blob/a292c8619be9904081712003e3872d446afb02a3/iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollObject.m#L850) expected the property's return type to be an `NSObject` derived type, but it received an `NSUInteger` which is a typedef'ed unsigned int. That's why I changed it to `NSNumber`, which is the same return type our `TiUIListViewProxy` uses for `sectionCount` [here](https://github.com/appcelerator/titanium_mobile/blob/a292c8619be9904081712003e3872d446afb02a3/iphone/Classes/TiUIListViewProxy.h#L15).